pulsar-broker/src/main/java/org/apache/pulsar/PulsarBrokerStarter.java

```diff
@@ -144,8 +144,11 @@ private static boolean argsContains(String[] args, String arg) {
                     workerConfig = WorkerConfig.load(starterArguments.fnWorkerConfigFile);
                 }
                 // worker talks to local broker
-                workerConfig.setPulsarServiceUrl("pulsar://127.0.0.1:" + brokerConfig.getBrokerServicePort());
-                workerConfig.setPulsarWebServiceUrl("http://127.0.0.1:" + brokerConfig.getWebServicePort());
+                boolean isTls = workerConfig.isUseTls();
+                workerConfig.setPulsarServiceUrl(String.format("pulsar://127.0.0.1:%s",
+                        (isTls ? brokerConfig.getBrokerServicePortTls() : brokerConfig.getBrokerServicePort()), args));
+                workerConfig.setPulsarWebServiceUrl(String.format("%s://127.0.0.1:%s", (isTls ? "https" : "http"),
+                        (isTls ? brokerConfig.getWebServicePortTls() : brokerConfig.getWebServicePort())));
```

Comment:
Does IP address URL cause hostname validation error?
Should we use `InetAddress.getLocalHost().getHostName()` or `advertisedAddress`?